### DISCUSSION TITLE
Pull a single relation field

### DISCUSSION THREAD

**Discussion by stratosgear** [pocketbase/pocketbase#2623] [Q&A] [answered]
I can successfully call:

http://my.domain/api/collections/calendarEvents/records?fields=id,title
[above call works]

But I would like to also do:

http://my.domain/api/collections/calendarEvents/records?fields=id,title,parent.color
[above call does NOT work]

where `parent` is a valid relationship to another table.

I know I could do:

http://my.domain/api/collections/calendarEvents/records?expand=parent
[above call works]

but that would pull the whole `parent` entity that it is very large and I do not need (plus it comes for multiple rows so the final response is huge)

Also something like:

http://my.domain/api/collections/calendarEvents/records?fields=id,title,parent.color&expand=parent
[above call does NOT work]

does not seem to help.

Am I missing a special invocation, or is this feature missing...?

I DID search for similar issues but couldn't find any (at least with the keywords that I used)

Thanks!

**Comment by stratosgear** [comment, marked as answer]
Ha, I had to play a little bit with some "undocumented?" incantations and I got this working:

http://my.domain/api/collections/calendarEvents/records?fields=id,title,expand.parent.color&expand=parent
[above call works]

I would consider this a perfect solution, unless I could also do:

http://my.domain/api/collections/calendarEvents/records?fields=*,expand.parent.color&expand=parent
[above call does not work]

but it seems "asterisk" does not expands all fields!   Is this worth a separate feature request...?

**Reply by ganigeorgiev** [reply]
That's correct, the `fields` parameter must have the same structure as the `items` json object when listing records. I'll consider documenting it better in the near future.

> but it seems "asterisk" does not expands all fields! Is this worth a separate feature request...?

I intentionally didn't add `*` support as it would be ambiguous (eg. should root level `*` also return the `expand` object?). Also keep in mind that the `fields` parameter is a generic and it is available for every API endpoint, not just for the records.

**Reply by stratosgear** [reply]
I am not sure I understand the implications of "fields" being a generic (in the context of go) but it *might* be worth it implementing for cases like the above.  I "could" manually expand the * to all the fields, in that particular case, but if I ever add another field to the collection I would have to also to remember to edit other unrelated queries.  And if you ever considered it I would expect * to refer to the native collection fields only, not the expand field., just like sql?

But I guess it all depends on your backlog.

You have a really cool project here! Thanks!
